src/mrsimulator/method/__init__.py

```diff
@@ -474,11 +474,12 @@ def get_transition_pathways(self, spin_system) -> List[TransitionPathway]:
             for item in segments
         ]
 
-    def summary(self, properties=None) -> pd.DataFrame:
+    def summary(self, properties=None, drop_constant_cols=False) -> pd.DataFrame:
```

Comment:
`summary` is now too complex. There may be an easier way of writing the method with the same logic, but that may restrict functionality down the line.

@DeepanshS Do you think it would be better to break summary into smaller functions or add a method to each Event class to get valid attributes?